src/main/java/com/blackducksoftware/integration/hub/alert/web/actions/ConfigActions.java

```diff
@@ -0,0 +1,94 @@
+import java.util.List;
+import java.util.Map;
+
+import org.springframework.data.jpa.repository.JpaRepository;
+
+import com.blackducksoftware.integration.exception.IntegrationException;
+import com.blackducksoftware.integration.hub.alert.datasource.entity.DatabaseEntity;
+import com.blackducksoftware.integration.hub.alert.web.ObjectTransformer;
+import com.blackducksoftware.integration.hub.alert.web.model.ConfigRestModel;
+
+public abstract class ConfigActions<D extends DatabaseEntity, R extends ConfigRestModel> {
+    public final Class<D> databaseEntityClass;
+    public final Class<R> configRestModelClass;
+    public final JpaRepository<D, Long> repository;
+    public final ObjectTransformer objectTransformer;
+
+    public ConfigActions(final Class<D> databaseEntityClass, final Class<R> configRestModelClass, final JpaRepository<D, Long> repository, final ObjectTransformer objectTransformer) {
+        this.databaseEntityClass = databaseEntityClass;
+        this.configRestModelClass = configRestModelClass;
+        this.repository = repository;
+        this.objectTransformer = objectTransformer;
+    }
+
+    public boolean doesConfigExist(final String id) {
+        return doesConfigExist(objectTransformer.stringToLong(id));
+    }
+
+    public boolean doesConfigExist(final Long id) {
+        return id != null && repository.exists(id);
+    }
+
+    public List<R> getConfig(final Long id) throws IntegrationException {
+        if (id != null) {
+            final D foundEntity = repository.findOne(id);
+            if (foundEntity != null) {
+                final R restModel = objectTransformer.transformObject(foundEntity, configRestModelClass);
+                if (restModel != null) {
+                    return Arrays.asList(restModel);
+                }
+            }
+            return Collections.emptyList();
+        }
+        final List<R> restModels = objectTransformer.transformObjects(repository.findAll(), configRestModelClass);
+        if (restModels != null) {
+            return restModels;
+        }
+        return Collections.emptyList();
+    }
+
+    public void deleteConfig(final String id) {
+        deleteConfig(objectTransformer.stringToLong(id));
+    }
+
+    public void deleteConfig(final Long id) {
+        repository.delete(id);
+    }
+
+    public D saveConfig(final R restModel) throws IntegrationException {
+        if (restModel != null) {
+            try {
+                D createdEntity = objectTransformer.transformObject(restModel, databaseEntityClass);
+                if (createdEntity != null) {
+                    createdEntity = repository.save(createdEntity);
+                }
+                return createdEntity;
+            } catch (final Exception e) {
+                throw new IntegrationException();
+            }
+        }
+        return null;
+    }
+
+    public abstract Map<String, String> validateConfig(R restModel);
+
+    public void customTriggers(final R restModel) {
```

Comment:
I understand why this needs to be empty having seen the rest of the PR, but we should definitely have a comment explaining it. I would also rename it to something more descriptive, along the lines of respondToConfigChanges(...).